source/libs/utils.ts

```diff
@@ -156,23 +156,33 @@ export function getLatestVersionTag(tags: string[]): string {
 	return latestVersion;
 }
 
-export function parseRoute(pathname: string): string[] {
+interface Pathname {
+	user: string;
+	repository: string;
+	route: string;
+	branch: string;
+	filePath: string;
+	toString: () => string;
+}
+export function parseRoute(pathname: string): Pathname {
 	const [user, repository, route, ...next] = pathname.replace(/^\/|\/$/g, '').split('/');
 	const parts = next.join('/');
-	const currentBranch = getCurrentBranch();
-	if (parts !== currentBranch && !parts.startsWith(currentBranch + '/')) {
+	const branch = getCurrentBranch();
+	if (parts !== branch && !parts.startsWith(branch + '/')) {
 		throw new Error('The branch of the current page must match the branch in the `pathname` parameter');
 	}
 
-	const filePath = parts.replace(currentBranch, '').replace(/^\//, '');
-	return [
-		'',
+	const filePath = parts.replace(branch, '').replace(/^\//, '');
+	return {
 		user,
 		repository,
 		route,
-		currentBranch,
-		filePath
-	];
+		branch,
+		filePath,
+		toString() {
+			return `/${this.user}/${this.repository}/${this.route}/${this.branch}/${this.filePath}`.replace(/\/$/, '');
```

Comment:
```suggestion
			return `/${user}/${repository}/${route}/${branch}/${filePath}`.replace(/\/$/, '');
```

Reply:
When I left out the `this` any replacement I made it did not change it in the toString()

Reply:
[![](https://i.kym-cdn.com/entries/icons/facebook/000/023/021/e02e5ffb5f980cd8262cf7f0ae00a4a9_press-x-to-doubt-memes-memesuper-la-noire-doubt-meme_419-238.jpg)](https://knowyourmeme.com/memes/la-noire-doubt-press-x-to-doubt)

Reply:
I tried it again.
Without the this it does not replace anything when calling toString()

Reply:
Heh you’re right, the object is changed but the constants aren’t. I wasn’t thinking about the replacements

Reply:
> Heh you’re right

It can happen... sometimes :smile: